### PR TITLE
Fail if resource is used without storage backend (fixes #538)

### LINKED ISSUE #538: AttributeError: 'Registry' object has no attribute 'storage'
I ran into this error message when I did  ```http -v http://localhost:6543/v0/champions --auth user:pass```

```
(env)[choclate@choclate contest]$ pserve development.ini --reload
Starting subprocess with file monitor
2015-11-02 23:39:57,102 INFO  [venusian][MainThread] contest 0.0 starting. 
Starting server in PID 4651.
serving on http://0.0.0.0:6543
2015-11-02 23:39:59,033 ERROR [venusian][waitress] "GET   /v0/champions" ? (? ms) 'Registry' object has no attribute 'storage' authn_type=BasicAuth; lang=None; uid=5741b99cc5b467903f9a392b092d59f6d227354acbca1834bdd1dea1dd350a8e; agent=HTTPie/0.9.2; exception=Traceback (most recent call last):
  File "/home/choclate/documents/projects/web/contest/env/lib/python3.5/site-packages/pyramid/tweens.py", line 21, in excview_tween
    response = handler(request)
  File "/home/choclate/documents/projects/web/contest/env/lib/python3.5/site-packages/pyramid/router.py", line 163, in handle_request
    response = view_callable(context, request)
  File "/home/choclate/documents/projects/web/contest/env/lib/python3.5/site-packages/pyramid/config/views.py", line 596, in __call__
    return view(context, request)
  File "/home/choclate/documents/projects/web/contest/env/lib/python3.5/site-packages/pyramid/config/views.py", line 329, in attr_view
    return view(context, request)
  File "/home/choclate/documents/projects/web/contest/env/lib/python3.5/site-packages/pyramid/config/views.py", line 305, in predicate_wrapper
    return view(context, request)
  File "/home/choclate/documents/projects/web/contest/env/lib/python3.5/site-packages/pyramid/config/views.py", line 245, in _secured_view
    return view(context, request)
  File "/home/choclate/documents/projects/web/contest/env/lib/python3.5/site-packages/pyramid/config/views.py", line 355, in rendered_view
    result = view(context, request)
  File "/home/choclate/documents/projects/web/contest/env/lib/python3.5/site-packages/pyramid/config/views.py", line 501, in _requestonly_view
    response = view(request)h
  File "/home/choclate/documents/projects/web/contest/env/lib/python3.5/site-packages/cornice/service.py", line 541, in wrapper
    ob = args['klass'](**params)
  File "/home/choclate/documents/projects/web/contest/env/lib/python3.5/site-packages/cliquet/resource/__init__.py", line 139, in __init__
    storage=request.registry.storage,
AttributeError: 'Registry' object has no attribute 'storage'; errno=None
2015-11-02 23:39:59,034 INFO  [venusian][waitress] "GET   /v0/champions" 500 (2 ms) request.summary authn_type=BasicAuth; lang=None; uid=5741b99cc5b467903f9a392b092d59f6d227354acbca1834bdd1dea1dd350a8e; agent=HTTPie/0.9.2; time=2015-11-02T23:39:59.033999; errno=999
```

This is a very small app with only one ressource and one schema. Last time I tried to used cliquet everything was fine. Is there a problem if I use virtualenv? There is no use of virtualenv in the doc, but it should be ok, right?  

I tried to add ```cliquet.storage_backend = cliquet.storage.memory``` in development.ini. I also tried to use python3.4.

The source code is available here: https://github.com/Atilla106/contest

Any ideas?

### PR DESCRIPTION
**Pull request by leplatrem**
@ametaireau r?

### CODE REVIEW

**Comment by Natim** [comment]
Great :)

File: cliquet/resource/__init__.py [hunk 1]
```diff
@@ -103,6 +104,12 @@ def callback(context, name, ob):
         # and call them from here when the @resource classes are being
         # scanned by venusian.
         config = context.config.with_package(info.module)
+
+        # Storage is mandatory for resources.
+        if not hasattr(config.registry, 'storage'):
+            error_msg = 'Mandatory storage backend missing from configuration.'
```
Comment (ametaireau) [inline cliquet/resource/__init__.py]: nit: "is missing"

**Comment by ametaireau** [comment]
Looks good to me, apart from the nit.

File: cliquet/tests/resource/test_viewset.py [hunk 2]
```diff
@@ -351,6 +352,16 @@ def setUp(self):
         self.viewset = FakeViewSet()
 
     @mock.patch('cliquet.resource.Service')
+    def test_register_fails_if_no_storage_backend_is_configured(self, *args):
+        venusian_callback = register_resource(
+            self.resource, viewset=self.viewset)
+        context = mock.MagicMock()
+        config = testing.setUp(settings=DEFAULT_SETTINGS)
+        context.config.with_package.return_value = config
+        with self.assertRaises(exceptions.ConfigurationError):
```
Comment (ametaireau) [inline cliquet/tests/resource/test_viewset.py]: Should we check that the correct error message is raised?

File: cliquet/tests/resource/test_viewset.py [hunk 3]
```diff
@@ -351,6 +352,19 @@ def setUp(self):
...
+        venusian_callback = register_resource(
+            self.resource, viewset=self.viewset)
+        context = mock.MagicMock()
+        config = testing.setUp(settings=DEFAULT_SETTINGS)
+        context.config.with_package.return_value = config
+        try:
+            venusian_callback(context, None, None)
+        except exceptions.ConfigurationError as e:
+            error = e
+        self.assertIn('storage backend is missing', error.message)
```
Comment (Natim) [inline cliquet/tests/resource/test_viewset.py]: In Python3 exception doesn't have the .message attribute.

Comment (ametaireau) [inline cliquet/tests/resource/test_viewset.py]: Apparently `error.message` doesn't exist in python 3.4.

Comment (Natim) [inline cliquet/tests/resource/test_viewset.py]: You may use ``str(e)`` instead.

Comment (Natim) [inline cliquet/tests/resource/test_viewset.py]: I am fixing it.

**Pull request merged**